### PR TITLE
Separated lint steps in CircleCI

### PR DESCRIPTION
**Pull request by kryswisnaskas**
**Description of change**
Separates lint steps in CircleCI

**How to test**

**Issue(s)**
Closes https://github.com/HHS/Head-Start-TTADP/issues/0

**Checklist**

- [ ] Meets issue criteria
- [x] Code tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -47,16 +47,24 @@ jobs:
           root: .
           paths:
             - .
-  lint_backend_and_frontend:
```
Comment (jasalisbury) [inline .circleci/config.yml:50]: I think you'll need to remove this job on line 101 in this file and add `lint_backend` and `lint_backend` jobs

**Review by SarahJaine — CHANGES_REQUESTED** [review]
Hey @kryswisnaskas, Can you explain a bit about why you want to separate the linting into two jobs? The whole linting job takes about 20 seconds usually, and of that only 2 seconds is spent on actual linting.  Spinning up the env and persisting the workspace is the time intensive part, and is repeated with each new job. I'm think splitting the linting into two jobs won't save us any time.  Here's an example ci run so you can compare times: https://app.circleci.com/pipelines/github/adhocteam/Head-Start-TTADP/186/workflows/9f8627b0-c8c6-42cd-a730-bf440fcd4eb1/jobs/504

**Comment by SarahJaine** [comment]
> Hey @kryswisnaskas, Can you explain a bit about why you want to separate the linting into two jobs? The whole linting job takes about 20 seconds usually, and of that only 2 seconds is spent on actual linting. Spinning up the env and persisting the workspace is the time intensive part, and is repeated with each new job. Here's an example ci run so you can compare times: https://app.circleci.com/pipelines/github/adhocteam/Head-Start-TTADP/186/workflows/9f8627b0-c8c6-42cd-a730-bf440fcd4eb1/jobs/504

I think I found it, where you trying to resolve this idea from Ryan >>
https://github.com/HHS/Head-Start-TTADP/pull/40#discussion_r488011376

If so, what he was requesting the linting be run as two commands.  What you did here was create two separate jobs. I'll add inline suggestions to your PR to demonstrate what I think he meant.

File: .circleci/config.yml [hunk 2]
```diff
@@ -47,16 +47,24 @@ jobs:
...
+  lint_backend:
```
Comment (SarahJaine) [inline .circleci/config.yml:50]:
```suggestion
  lint_backend_and_frontend:
```
We'd leave this job name alone.

File: .circleci/config.yml [hunk 3]
```diff
@@ -47,16 +47,24 @@ jobs:
...
     executor: docker-executor
     steps:
       - attach_workspace:
           at: .
       - run:
-          name: Lint backend and frontend
-          command: yarn run lint:ci && cd frontend && yarn run lint:ci
+          name: Lint backend
+          command: yarn run lint:ci
```
Comment (SarahJaine) [inline .circleci/config.yml:57]:
```suggestion
          command: yarn run lint:ci
      - run:
          name: Lint frontend
          command: yarn --cwd frontend run lint:ci
```
Add a step to run the command for linting the frontend.

File: .circleci/config.yml [hunk 4]
```diff
@@ -47,16 +47,24 @@ jobs:
...
       - store_artifacts:
           path: reports
+  lint_frontend:
+    executor: docker-executor
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Lint frontend
+          command: cd frontend && yarn run lint:ci
```
Comment (SarahJaine) [inline .circleci/config.yml:67]:
```suggestion
```
Get rid of this whole new job block, you don't need it.

**Review by SarahJaine — CHANGES_REQUESTED** [review]
I think this is what Ryan was suggesting.  You probably need to double check my line indentation here. :)

File: .circleci/config.yml (hunk 4 above)
Comment (kryswisnaskas) [reply .circleci/config.yml:67]: Yup, @SarahJaine , good point. He meant steps not jobs.

**Review by jasalisbury — APPROVED** [review]
:+1:

**Review by SarahJaine — APPROVED** [review]
Excellent, thanks for making those changes!

**Pull request merged**